docs/VTPM.md

```diff
@@ -1,68 +1,7 @@
-# VTPM Server
+# VTPM
 
-VTPM (located in `pkg/vtpm`) is a server listening on port `8877` in EVE, exposing limited functionality of the TPM to the clients. VTPM allows clients to execute [tpm2-tools](https://github.com/tpm2-software/tpm2-tools) binaries from a list of [hardcoded options](https://github.com/lf-edge/eve/blob/883547fe7978550a30e4389aac24d562d1dae105/pkg/vtpm/src/server.cpp#L58).
+*(if you're looking for old VTPM documents, please refer to [PTPM](docs/PTPM.md))*
 
-## Packet Structure
+Virtual TPM container integrates SWTPM with QEMU, in order to emulate a full Virtual TPM for running VMs and bare-metal containser. It create a SWTPM instance per VM. The SWTPM instance is configured to use a Unix Domain Socket as a communication line, by passing the socket path to the QEMU virtual TPM configuration, QEMU automatically creates a virtual TPM device for the VM which is accessible like a normal TPM under `/dev/tpm*`.
```

Comment:
> Please clarify whether this is only a TPM 2.,0 (/hence /dev/tpmrm*) or a combined TPM 1.2 and 2.0.

added. 

> Also, today app instance need to opt-in to use the virtual TPM (using eve-tools etc).
In this new world do you think we have provide the TPM device to all deployed containers and VMs, or can that result in breakage and we should have the user choose when deploying apps?

In theory it should not result in breakage, since the only change is that now VM can see a tpm device in `/dev`, the old vtpm (now ptpm) is still running in the vtpm container for foreseeable future, I just renamed the binary.

But, I'm working one eden test for both swtpm and az-iot to make sure in there is no breakage in practice either.

Reply:
We definitelhy need to test with (different versions of) the azure IoT Edge runtime to make sure it doesn't automatically switch over or otherwise get confused by the /dev/tpm* devices showing up in the guest VM.